core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlUtils.java

```diff
@@ -68,43 +52,39 @@ private boolean isOperandAnalyticalInFollowingProject(RelNode rel, Integer rexOp
   }
 
   /** Returns whether an Analytical Function is present in filter condition. */
-  protected boolean hasAnalyticalFunctionInFilter(Filter rel) {
-    Integer rexOperandIndex = null;
-    RexNode filterCondition = rel.getCondition();
-    if (filterCondition instanceof RexCall) {
-      for (RexNode conditionRex : ((RexCall) filterCondition).getOperands()) {
-        if (conditionRex instanceof  RexLiteral) {
-          continue;
-        }
-
-        List<RexNode> inputRefRexList = new ArrayList<>();
-        List<RexNode> rexOperandList =
-            getOperandsOfTypeRexInputRefFromRexNode(conditionRex, inputRefRexList);
+  protected boolean hasAnalyticalFunctionInFilter(Filter rel, RelNode input) {
+    try {
+      rel.getCondition().accept(new AnalyticalFunFinder(true, input));
+      return false;
+    } catch (Util.FoundOne e) {
+      return true;
+    }
+  }
 
-        for (RexNode rexOperand : rexOperandList) {
-          if (rexOperand instanceof RexInputRef) {
-            rexOperandIndex = ((RexInputRef) rexOperand).getIndex();
-            if (isOperandAnalyticalInFollowingProject(rel, rexOperandIndex)) {
-              return true;
-            }
-          }
-        }
-      }
+  protected boolean hasAnalyticalFunctionInJoin(RelNode input) {
+    if (input instanceof LogicalJoin && input.getInput(0) instanceof Project) {
+      return isAnalyticalFunctionPresentInProjection((Project) input.getInput(0));
     }
     return false;
   }
 
-  /** Returns whether any Analytical Function (RexOver) is present in projection. */
+  /* Returns whether any Analytical Function (RexOver) is present in projection.*/
   protected boolean isAnalyticalFunctionPresentInProjection(Project projectRel) {
     for (RexNode currentRex : projectRel.getProjects()) {
       if (currentRex instanceof RexOver) {
```

Comment:
can you modify this method and use the existing method inside for loop as below.

protected boolean isAnalyticalFunctionPresentInProjection(Project projectRel) {
    for (RexNode currentRex : projectRel.getProjects()) {
        if (isAnalyticalRex(currentRex)) {
            return true;
        }
    }
    return false;
}